drivers/sensor/lsm6dso/lsm6dso.c

```diff
@@ -773,6 +773,52 @@ static int lsm6dso_channel_get(const struct device *dev,
 	return 0;
 }
 
+#if defined(CONFIG_LSM6DSO_ENABLE_FIFO)
```

Comment:
This might be my limited understanding of C, but I find it strange that the function is defined behind `#if defined(CONFIG_LSM6DSO_ENABLE_FIFO)` here, but in the header file and usages it's not guarded by the same constraint. Won't this cause a compilation issue if `CONFIG_LSM6DSO_ENABLE_FIFO` is not defined?

Reply:
this guards static functions that are not defined in the header and only ever used if CONFIG_LSM6DSO_ENABLE_FIFO is enabled.
Agreed this looks ugly but is sort of the "zephyr way" to go.

Reply:
Oh I see it now, sorry, github did not show me the other `#if` because it was too far from the function calls.